### ISSUE TITLE
Cannot `@show` AxisTensor

### ISSUE THREAD

**Issue by charleskawczynski** [CliMA/ClimaCore.jl#768]
## Describe the bug

It seems as though we cannot `@show` `AxisTensor`s:

## To Reproduce

```julia
using ClimaCore: Geometry
using StaticArrays

T = Geometry.AxisTensor{Float64, 2, Tuple{Geometry.LocalAxis{(1, 2)}, Geometry.CovariantAxis{(1, 2)}}, SMatrix{2, 2, Float64, 4}};
components = SMatrix{2, 2, Float64, 4}([4.0 0.0; 0.0 5.0]);
axes_T = (Geometry.LocalAxis{(1, 2)}(), Geometry.CovariantAxis{(1, 2)}());
ats = T(axes_T, components);
@show ats
```

**Comment by charleskawczynski** [comment]
Results in a method error:

```julia
julia> @show ats
ERROR: MethodError: no method matching _show_nonempty(::IOBuffer, ::ClimaCore.Geometry.AxisTensor{Float64, 2, Tuple{ClimaCore.Geometry.LocalAxis{(1, 2)}, ClimaCore.Geometry.CovariantAxis{(1, 2)}}, SMatrix{2, 2, Float64, 4}}, ::String, ::Bool, ::Tuple{ClimaCore.Geometry.LocalAxis{(1, 2)}, ClimaCore.Geometry.CovariantAxis{(1, 2)}})
Closest candidates are:
  _show_nonempty(::IO, ::AbstractMatrix, ::String, ::Bool, ::Tuple{AbstractUnitRange, AbstractUnitRange}) at /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/base/arrayshow.jl:413
  _show_nonempty(::IO, ::AbstractMatrix, ::String) at /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/base/arrayshow.jl:410
  _show_nonempty(::IO, ::AbstractArray, ::String) at /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/base/arrayshow.jl:462
  ...
Stacktrace:
 [1] _show_nonempty(io::IOBuffer, X::ClimaCore.Geometry.AxisTensor{Float64, 2, Tuple{ClimaCore.Geometry.LocalAxis{(1, 2)}, ClimaCore.Geometry.CovariantAxis{(1, 2)}}, SMatrix{2, 2, Float64, 4}}, prefix::String)
   @ Base ./arrayshow.jl:410
 [2] show(io::IOBuffer, X::ClimaCore.Geometry.AxisTensor{Float64, 2, Tuple{ClimaCore.Geometry.LocalAxis{(1, 2)}, ClimaCore.Geometry.CovariantAxis{(1, 2)}}, SMatrix{2, 2, Float64, 4}})
   @ Base ./arrayshow.jl:486
 [3] sprint(f::Function, args::ClimaCore.Geometry.AxisTensor{Float64, 2, Tuple{ClimaCore.Geometry.LocalAxis{(1, 2)}, ClimaCore.Geometry.CovariantAxis{(1, 2)}}, SMatrix{2, 2, Float64, 4}}; context::Nothing, sizehint::Int64)
   @ Base ./strings/io.jl:114
 [4] repr(x::ClimaCore.Geometry.AxisTensor{Float64, 2, Tuple{ClimaCore.Geometry.LocalAxis{(1, 2)}, ClimaCore.Geometry.CovariantAxis{(1, 2)}}, SMatrix{2, 2, Float64, 4}}; context::Nothing)
...
```

**Comment by charleskawczynski** [comment]
I ran into this when trying to fix #720..
